### DISCUSSION TITLE
Installation Trouble

### DISCUSSION THREAD

**Discussion by welt3r** [diasurgical/DevilutionX#2372] [General]
I've followed the instructions in the Wiki for Extracting the .MPQs from the GoG installer. When I go to use innoextract however, the output says it could not open the file. Here's the input and output: 

$ innoextract -I DIABDAT.MPQ -I hellfire.mpq -I hfmonk.mpq -I hfmusic.mpq -I hfvoice.mpq 'setup_diablo_1.09_hellfire_v2_(30038).exe' && mv hellfire/h*.mpq ./ && rmdir hellfire

Could not open file "setup_diablo_1.09_hellfire_v2_(30038).exe"
Done with 1 error.

Can anyone help me resolve this? I followed the instructions and even restarted from the beginning, reinstalling everything. For reference, I am running Mac OS 10.14.6.

Apologies if this post doesn't fit the format or isn't appropriate. I'm new to this.

**Comment by ephphatha** [comment]
Those instructions are for extracting the data files from the offline installer, if you've already installed the game using GoG Galaxy or running the installer the MPQs are extracted as part of the install process.

You can find them by opening the install folder:
![2021-07-13T160936_GOG_Galaxy_2 0 37](https://user-images.githubusercontent.com/357684/125400198-f8c33480-e3f4-11eb-9f3b-2995cb8b02a9.png)
DIABDAT.mpq is in the folder that appears and the hellfire MPQs are in ./hellfire (this is on windows, the mac client may be different).

Otherwise the error sounds like the command is not running in the same folder as you've saved the installer. Can you confirm that the file setup_diablo_1.09_hellfire_v2_(30038).exe exists by running `ls setup_diablo*` and confirming it lists at least one file?

**Reply by welt3r** [reply]
Thanks. I haven't installed the game as the GoG version is only compatible with Windows. I downloaded the offline installer and the file in my downloads folder. However, I ran the script you suggest, ls setup_diablo* and I get the result: No such file or directory. Strange. Do I need to place the installer in a specific folder somewhere for the command to find it?

**Reply by AJenbo** [reply]
You need to give the correct path to the installer relative to what folder your terminal is currently operating in. You can see the current path before the `$` in the input line.
Most probably you are in the home folder and since the installer is in the Downloads folder you need to either prefix the path with `Downloads/` or move to that folder via the command `cd` (change directory). You could also move the installer to the home folder or provide the absolute path instead.

![image](https://user-images.githubusercontent.com/204594/125501674-65c1e0ab-d509-4d3b-9a96-ae2b63c2ce1e.png)

**Reply by welt3r** [reply]
Thanks Ajenbo, that worked. Terminal was operating in the home folder, so moving the installer into the folder resolved the issue. 

Alas, I am now so close to being able to play, but can't seem to get the mpq files to the proper location. I followed the wiki directions, but entering the command: cp *.mpq ~/Library/Application\ Support/diasurgical/devilution  appears to do nothing. This makes sense to me though as after installing devilutionx, all I have is an icon in my applications folder and nothing new in my application support folder. By right-clicking the devilutionx icon, I can "show package contents". I tried to add the mpq files to devilutionx by copying and pasting them into this contents folder, but when I try to run it, I get an error indicating "Unable to open main data archive"

**Reply by AJenbo** [reply]
Make sure you are looking in your users application support folder (hidden by default) and not the system application support folder.

The application should have created the folders the frist time you run it. Putting the mpq in the application content is not the correct thing to do.

**Reply by welt3r** [reply]
Thank you! Just got it working. I was looking in the system application support folder, not the user. I changed diabdat.mpq from upper to lower case and ran the script in the wiki and now it's running. Appreciate your help!
